open_spiel/algorithms/infostate_tree.cc

```diff
@@ -491,12 +491,57 @@ std::shared_ptr<InfostateTree> MakeInfostateTree(
     const Game& game, Player acting_player,
     int max_move_limit) {
   return std::shared_ptr<InfostateTree>(new InfostateTree(
-      game, acting_player, max_move_limit));
+      {game.NewInitialState().get()}, /*chance_reach_probs=*/{1.},
+      game.MakeObserver(kInfoStateObsType, {}),
+      acting_player, max_move_limit));
+}
+
+std::shared_ptr<InfostateTree> MakeInfostateTree(
+    const std::vector<InfostateNode*>& start_nodes,
+    int max_move_ahead_limit) {
+  std::vector<const InfostateNode*> const_nodes(
+      start_nodes.begin(), start_nodes.end());
+  return MakeInfostateTree(const_nodes, max_move_ahead_limit);
+}
+
+std::shared_ptr<InfostateTree> MakeInfostateTree(
+    const std::vector<const InfostateNode*>& start_nodes,
```

Comment:
Are the pointers in the vector managed elsewhere? If yes, then it's OK, otherwise please use some smart pointer type.

Reply:
Yes they are managed by the another tree where they may come from (when we want to construct tree continuation based on a set of nodes)